server/src/main/java/io/crate/role/RoleManager.java

```diff
@@ -64,9 +64,16 @@ CompletableFuture<Long> createRole(String roleName,
      *
      * @param roleName of the existing role to modify
      * @param newHashedPw new password; if null the password is removed from the role
+     * @param newJwtProperties new jwt properties. if null properties are removed from the role
+     * @param resetPassword hints how to treat NULL password: as not provided and supposed to be kept or explicitly set to NULL.
+     * @param resetJwtProperties hints how to treat NULL jwt: as not provided and supposed to be kept or explicitly set to NULL.
      * @return 1 if the role has been updated, otherwise a failed future.
      */
-    CompletableFuture<Long> alterRole(String roleName, @Nullable SecureHash newHashedPw);
+    CompletableFuture<Long> alterRole(String roleName,
+                                      @Nullable SecureHash newHashedPw,
+                                      @Nullable JwtProperties newJwtProperties,
+                                      boolean resetPassword,
+                                      boolean resetJwtProperties);
```

Comment:
I wonder if it would be worth adding some kind of `Assignment<T>`  to pair the reset flag and the value.

E.g. have `Assigment.absent()` and `Assignment.newValue(null | <concrete_value>)` - in a way similar to `Optional` except that you can have a `null` in the `present` case